apps/studio/src/server/modules/page/page.service.ts

```diff
@@ -42,3 +46,49 @@ export const createDefaultPage = ({
     }
   }
 }
+
+export const addNewVersion = async (siteId: number, pageId: number) => {
```

Comment:
suggestion: I think this procedure should be kept within versions service and is an implementation that page service should not be concerned with.

Main reason is because I see page service as the one responsible for returning the draft and published page contents. The versions service is the one that supports the page service by providing the latest version, and how this information is stored is the responsibility of the versions service. So if versions service does any incrementing of the version number and stuff, that is up to it.

Reply:
yup was pending refactor, extracted now